pyomo/solvers/plugins/solvers/scip_direct.py

```
@@ -0,0 +1,838 @@
+        if self._objective is not None:
+            for var in self._vars_referenced_by_obj:
+                self._referenced_variables[var] -= 1
+            self._vars_referenced_by_obj = ComponentSet()
+            self._objective = None
+
+        if obj.active is False:
+            raise ValueError("Cannot add inactive objective to solver.")
+
+        if obj.sense == minimize:
+            sense = "minimize"
+        elif obj.sense == maximize:
+            sense = "maximize"
+        else:
+            raise ValueError("Objective sense is not recognized: {0}".format(obj.sense))
+
+        scip_expr, referenced_vars = self._get_expr_from_pyomo_expr(
+            obj.expr, self._max_obj_degree
+        )
+
+        for var in referenced_vars:
+            self._referenced_variables[var] += 1
+
+        self._solver_model.setObjective(scip_expr, sense=sense)
+        self._objective = obj
+        self._vars_referenced_by_obj = referenced_vars
+
+        self._needs_updated = True
+
+    def _postsolve(self):
+        # the only suffixes that we extract from SCIP are
+        # constraint duals, constraint slacks, and variable
+        # reduced-costs. scan through the solver suffix list
+        # and throw an exception if the user has specified
+        # any others.
+        extract_duals = False
+        extract_slacks = False
+        extract_reduced_costs = False
+        for suffix in self._suffixes:
+            flag = False
+            if re.match(suffix, "dual"):
+                extract_duals = True
+                flag = True
+            if re.match(suffix, "slack"):
+                extract_slacks = True
+                flag = True
+            if re.match(suffix, "rc"):
+                extract_reduced_costs = True
+                flag = True
+            if not flag:
+                raise RuntimeError(
+                    "***The scip_direct solver plugin cannot extract solution suffix="
+                    + suffix
+                )
```

Design question: is it better to raise an error or to log (or print loudly) that other requests are being ignored?

(Note: I don't have a strong preference either way.)

---

I would lean slightly towards the print loudly option, but can see the potential danger for users. As a larger preference though, I would like to stick to similar behavior with the other interfaces.